src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReferenceResolver.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// This class defines how the <see cref="JsonSerializer"/> deals with references on serialization and deserialization.
+    /// Defines the core behavior of preserving references on serialization and deserialization.
+    /// </summary>
+    public abstract class ReferenceResolver
+    {
+        /// <summary>
+        /// Adds an entry to the bag of references using the specified id and value.
+        /// This method gets called when an $id metadata property from a JSON object is read.
+        /// </summary>
+        /// <param name="referenceId">The identifier of the respective JSON object or array.</param>
+        /// <param name="value">The value of the respective CLR reference type object that results from parsing the JSON object.</param>
+        public abstract void AddReference(string referenceId, object value);
+
+        /// <summary>
+        /// Gets the reference identifier of the specified value if exists; otherwise a new id is assigned.
+        /// This method gets called before a CLR object is written so we can decide whether to write $id and enumerate the rest of its properties or $ref and step into the next object.
+        /// </summary>
+        /// <param name="value">The value of the CLR reference type object to get an id for.</param>
+        /// <param name="alreadyExists">`true` if a reference to value already exists; otherwise, `false`.</param>
```

Comment:
I think we should lean towards [documentation guidelines](https://github.com/dotnet/dotnet-api-docs/wiki/Returns:-Methods) since it will be corrected anyway when it gets ported to dotnet-api-docs repo.

Reply:
Do the guidelines say we shouldn't use syntax tags? Why wait to get the same feedback from the docs team?

Reply:
>Do the guidelines say we shouldn't use syntax tags?

No, you are right, we should use tag syntax here, however on remarks we do use backticks to reference langwords, see https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System.IO/FileStream.xml#L340

> Why wait to get potentially the same feedback from the docs team?

I was referring to your suggestion that includes "Set to" at the beginning of the sentence which was out of the docs guidelines; I think we must stick as much as we can to them to alleviate the amount of corrections needed when the docs are published.

Thanks for catching the lack of tag syntax.